packages/compiler/src/server/serverlib.ts

```diff
@@ -339,6 +343,165 @@ export function createServer(host: ServerHost): Server {
     }
   }
 
+  async function updateImportsOnFileChange(changes: FileEvent[]): Promise<void> {
+    const tspChanges = changes.filter((change) => change.uri.endsWith(".tsp"));
+    const filePathChangedMap = new Map<string, string>();
+
+    if (tspChanges.length === 0 || tspChanges.length % 2 !== 0) {
+      log({
+        level: "debug",
+        message:
+          "The arguments are empty or do not contain pairs of TSP files marked for deletion(type=3) and creation(type=1)",
```

Comment:
it feels a little overlogging. Rename is just a small part of file changes. Please remove unnecessary log.